addons/web/static/tests/fields/relational_fields_tests.js

```diff
@@ -2026,36 +2027,163 @@ QUnit.module('relational_fields', {
-            form.$('.ui-sortable-handle').eq(1),
+            form.$('.ui-sortable-handle').eq(2),
             form.$('tbody tr').first(),
             {position: 'top'}
         );
 
         assert.verifySteps(["0", "1", "2"],
             "sequences values should be incremental starting from the previous minimum one");
 
-        assert.strictEqual(form.$('td.o_data_cell:not(.o_handle_cell)').text(), "blipMy little Foo Valueyop",
+        assert.strictEqual(form.$('td.o_data_cell:not(.o_handle_cell)').text(), "kawayopblip",
             "should have the 3 rows in the new order");
 
         form.$buttons.find('.o_form_button_save').click();
-        assert.strictEqual(form.$('td.o_data_cell:not(.o_handle_cell)').text(), "blipMy little Foo Valueyop",
+        assert.strictEqual(form.$('td.o_data_cell:not(.o_handle_cell)').text(), "kawayopblip",
             "should still have the 3 rows in the new order");
 
         form.destroy();
     });
 
+    QUnit.test('onchange for embedded one2many with handle widget', function (assert) {
+        assert.expect(2);
+
+        this.data.partner.records[0].turtles = [1, 2, 3];
+        var partnerOnchange = 0;
+        this.data.partner.onchanges = {
+            turtles: function (obj) {
+                partnerOnchange++;
+            },
+        };
+        var turtleOnchange = 0;
+        this.data.turtle.onchanges = {
+            turtle_int: function (obj) {
+                turtleOnchange++;
+            },
+        };
+
+        var form = createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch:'<form string="Partners">' +
+                    '<sheet>' +
+                        '<notebook>' +
+                            '<page string="P page">' +
+                                '<field name="turtles">' +
+                                    '<tree default_order="turtle_int">' +
+                                        '<field name="turtle_int" widget="handle"/>' +
+                                        '<field name="turtle_foo"/>' +
+                                    '</tree>' +
+                                '</field>' +
+                            '</page>' +
+                        '</notebook>' +
+                    '</sheet>' +
+                 '</form>',
+            res_id: 1,
+        });
+
+        form.$buttons.find('.o_form_button_edit').click();
+
+        // Drag and drop the fourth line in second position
+        testUtils.dragAndDrop(
+            form.$('.ui-sortable-handle').eq(2),
+            form.$('tbody tr').first(),
+            {position: 'top'}
+        );
+
+        assert.strictEqual(turtleOnchange, 3, "should trigger on onchange per line");
+        assert.strictEqual(partnerOnchange, 1, "should trigger only one onchange on the parent");
+
+        form.destroy();
+    });
+
+    QUnit.test('embedded one2many with handle widget with minimum setValue calls', function (assert) {
```

Comment:
Where does this check the number of setValue calls? Implicitly as part of the number of assertions expected?

Reply:
yes because we check field_changed